server/worker_pool/worker_processor.py

```diff
@@ -160,6 +195,41 @@ async def _check_for_work(self):
         except Exception as e:
             logger.error(f"Worker {self.worker_id}: Error checking for work: {e}")
 
+    async def _check_memory_threshold(self):
+        """Check memory usage and trigger restart if needed"""
+
+        try:
+            settings = Settings()
+
+            # Get current memory without logging (to avoid spam)
+            try:
```

Comment:
I think this can be combined into a single utility function to follow DRY

Reply:
done